javaslang/src/main/java/javaslang/collection/LinearSeq.java

```diff
@@ -348,5 +388,20 @@ default int segmentLength(Predicate<? super T> predicate, int from) {
             return null;
         }
     }
+    interface Search {
+        static <T> int linearSearch(LinearSeq<T> seq, Function1<T, Integer> comparison) {
```

Comment:
Accordingly this could be replaced by `ToIntFunction`. I think it is a only a slight improvement but then we got the most out of it.

I was recently asked why I do not always use Javaslang Functions myself (within Javaslang). For consumers I try to use the most general function, mostly JDK API, like Function, BiFunction, etc. Producers return the most specfic types, E.g. Seq extends `Function1<Integer, T>` instead of `Function<Integer, T>`. Could be also `IntFunction`. These @FunctionalInterfaces drive me crazy :-) I wished there were no primitive types in Java and that Function1 and Function were the same.